### ISSUE TITLE
"Areas de ponderacao" para Censo 2000 (e anteriores)

### ISSUE THREAD

**Issue by lucasmation** [ipeaGIT/geobr#152]
Voces sabem desde quando existem areas de ponderacao nos Censos brasileiros?

Acho que para o Censo 2000 existe, mas estou tendo dificuldade em achar. Se voces tiverem seria interessante   colocar no pacote.

**Comment by rafapereirabr** [comment]
Oi Lucas. O ftp do IBGE só tem os dados de área de ponderação de 2010, mas se voce encontrar avisa que a gente inclui. Acho que a maneira mais facil seria contactar o pessoal do IBGE para perguntar

[ftp](ftp://geoftp.ibge.gov.br/recortes_para_fins_estatisticos)  ftp://geoftp.ibge.gov.br/recortes_para_fins_estatisticos

**Comment by lucasmation** [comment]
Valeu Rafa, vou perguntar la e aviso aqui. Vc sabe desde quando existem areas de ponderacao?

**Comment by lucasmation** [comment]
Respondendo minha propria pergunta, segundo o CEM existem para 2000 e 2010.

**Comment by rafapereirabr** [comment]
Sim, existem areas de ponderacao para o censo 2000, mas nao acho os shapes em lugar algum. e alguem tiver noticias de onde achar esses dados a gente pode reabrir o issue e incluir a base oo geobr. Fechando esse issue por enquanto.

**Comment by peredrozende** [comment]
Olá, Rafael! Cruzei com essa issue enquanto buscava a definição das áreas de ponderação de 2000 e encontrei uma solução em seguida. O IBGE disponibiliza os setores de 2000 no geoftp e a composição das áreas de ponderação no ftp. Fica no zip de [documentação dos microdados](https://ftp.ibge.gov.br/Censos/Censo_Demografico_2000/Microdados/1_Documentacao_20170908.zip), no caminho Arquivos Auxiliares/Composicao das Areas de Ponderacao.txt. Com algumas operações de geometria é possível reconstituir o perímetro das APs. Não programo em R, se não eu mesmo faria a contribuição.

**Comment by rafapereirabr** [comment]
Oi Pedro, muito obrigado pela sugestão! Seria possível fazermos isso sim para gerar as áreas de ponderação do Censo 2000. Vou reabrir esse issue para fazermos isso em algum momento. Mas confesso q isso deve demorar por falta de braços e pernas aqui no momento

**Comment by thiagocalm** [comment]
Ei @rafapereirabr e @peredrozende! Me deparei com essa issue por um interesse particular nestes shapes também. Recentemente fiz o contato com o Claudio Stenner (da parte de geociências do IBGE) em um evento, momento em que perguntei sobre (i) existência desses shapes para as Áreas de Ponderação para 2000; e (ii) existência de shapes compatibilizados para AP entre 2000 e 2010.
Naquele momento ele deu um retorno de que estavam fazendo testes internos ao IBGE sobre isso, mas sem prazo para colocarem no ar. Não sei em que pé está agora.

Com isso, queria incluir nessa issue esta nova tarefa também de tentar fazer a compatibilização das APs entre 2000 e 2010, além de compor a de 2000, seguindo a sugestão do @peredrozende.

Se ninguém tiver feito isso ainda, posso me juntar a você @peredrozende ou outra pessoa para tentarmos pensar nisso.

**Comment by peredrozende** [comment]
Olá, @thiagocalm!! :)
Legal saber que estão buscando uma solução "mais oficial" no IBGE!
Eu tive que fazer essa compatibilização para um mapeamento e acabei identificando alguns problemas difíceis de contornar. Pra reconstituir as APs de 2000 bastou unir as geometrias de setores de 2000 com os códigos das áreas de ponderação e dissolvê-las. Porém, pra compatibilizar com 2010, precisa de um trabalho quase artesanal. Trabalhando só com a RMSP, identifiquei estes problemas:

- Alguns municípios tiveram os perímetros de suas APs completamente refeitos de 2000 para 2010, sem contornos coincidentes;
- Alguns municípios (ex: Mogi das Cruzes) têm setores de 2000 muito distorcidos em relação à malha urbana, o que dificulta muito o processo de compatibilização geométrica - não dá pra ter clareza da intenção de quem fez o desenho;
- Os contornos das APs de 2000 e 2010 ficam bem desalinhados, mesmo quando coincidentes. Precisei realizar várias operações de geometria e exclusão por algumas métricas, excluir área viária, etc, e mesmo assim tive que passar um pente fino no olho, cruzando com imagens de satélite pra lidar com desalinhamentos espúrios em áreas não habitadas.

Enfim, pra fazer a compatibilização de forma puramente geométrica precisa ter um nível de detalhamento mínimo nos contornos, ou tudo acaba sendo agregado na mesma unidade. Como, apesar dessas ressalvas, as mudanças principais são basicamente de desmembramento e união, a lógica que usei foi de criar uma terceira malha, de "mínimos perímetros comuns" entre os dois censos, e então reagregar os dados de 2000 e 2010 pra essa nova malha. Quando se interseciona as malhas dos censos com as geometrias já tratadas, dá pra modelar a interseção como um grafo desconexo, onde cada nó é uma AP (de 2000 ou 2010) e cada aresta é uma sobreposição 2000-2010. Aí cada componente do grafo resultante é um mínimo perímetro comum. 
Temo que seja um trabalho difícil de replicar no atacado, mas deve ser possível fazer com cuidado para algumas RMs, se vocês acharem que faz sentido. Posso ajudar com o desenho da metodologia e eventualmente com algum código. Já a malha de APs 2000 dá pra garantir que é possível.

**Comment by rafapereirabr** [comment]
Oi pessoal. Temos um colega do Ipea, o @lucasmation, que criou um metodo com script em R para ver compatibilização de setores censitários entre os censos de 2000 e 2010. Isso permite criar o equivalente às 'áreas minimas comparáveis' só para setores censitários. Estou aguardando ele publicar o trabalho com a metodologia para podermos incluir isso no **geobr**. O Lucas usou uma lógica de grafos, algo parecido com o que o @peredrozende fez.

Com esse trabalho do Lucas, seria possivel recriar áreas de ponderação do censo 2000, mas essas coisas são sempre mais complicadas na prática do que a gente consegue antecipar.

O ideal mesmo seria aguardar a equipe do IBGE gerar os arquivos oficiais das areas de ponderação de 2000 (e corrigir os setores censitarios de 2000 tambem, que tem varios problemas).

**Comment by phchavesmaia** [comment]
Oi @rafapereirabr. Me envolvi em um projeto recentemente e, fortuitamente, me deparei com essa issue. O @lucasmation já publicou o trabalho em questão? Pergunto, pois seria um ferramental que me ajudaria substancialmente.
